Lib/Src/CParser.cpp

```diff
@@ -0,0 +1,309 @@
+{
+    // Reset command variables
+    m_status = IDLE;
+    m_argument_counter = 0;
+    m_command_name = "";
+
+    // Get tokens from string
+    getTokens(string);
+
+    for (uint16_t i = 0; i < m_token_counter; i++)
+    {
+        token_t current_token = m_tokens[i];
+
+        switch (current_token.type)
+        {
+            case BEGIN_OPERATOR:
+                if (m_status == IDLE)
+                {
+                    m_status = BEGIN_KEY;
+                }
+                else
+                {
+                    m_status = ERROR_NESTED_JSON;
+                }
+                break;
+            case STRING_LITERAL:
+                if (m_status == BEGIN_KEY)
+                {
+                    // Store key
+                    m_command_name = m_tokens[i].text;
+                    m_status = END_KEY;
+                }
+                break;
+            case START_ARRAY_OP:
+                if (m_status == BEGIN_VALUE)
+                {
+                    m_status = BEGIN_ARRAY;
+                }
+                break;
+            case ARGUMENT_DELIMITER:
+                if (m_status == BEGIN_ARRAY || m_status == VALUE_ADDED)
+                {
+                    m_status = ADD_VALUE;
+                }
+                break;
+            case KEY_VALUE_DIV:
+                if (m_status == END_KEY)
+                {
+                    m_status = BEGIN_VALUE;
+                }
+                break;
+            case INTEGER:
+            case FLOAT:
+                if (m_status == BEGIN_VALUE)
+                {
+                    float number = std::stof(m_tokens[i].text.c_str());
+                    m_arguments[m_argument_counter] = number;
+                    m_argument_counter++;
+                    m_status = END_VALUES;
+                }
+                else if (m_status == BEGIN_ARRAY || m_status == ADD_VALUE)
+                {
+                    if (m_argument_counter < MAX_ARGUMENT_COUNT)
+                    {
+                        float number = std::stof(m_tokens[i].text.c_str());
+                        m_arguments[m_argument_counter] = number;
+                        m_argument_counter++;
+                        m_status = VALUE_ADDED;
+                    }
+                    else
+                    {
+                        m_status = ERROR_MAX_ARGS;
+                    }
+                }
+                break;
+            case END_ARRAY_OP:
+                if (m_status == VALUE_ADDED)
+                {
+                    m_status = END_VALUES;
+                }
+                break;
+            case END_OPERATOR:
+                if (m_status == END_VALUES)
+                {
+                    m_status = COMMAND_OK;
+                }
+                break;
+            default:
+                m_status = ERROR;
+        }
+    }
+
+    return m_status;
+}
+
+/**
+ * @brief Process string to get tokens
```

Comment:
What does it mean `tokenised`? This header needs a lot more explanation. You should assume reader does not know how JSON works.